chiselFrontend/src/main/scala/Chisel/Assert.scala

```diff
@@ -28,32 +29,36 @@ object assert { // scalastyle:ignore object.name
     * defs need to be compiled first and the SBT project is not set up to do
     * that
     */
-  def apply(cond: Bool, message: String): Unit = macro apply_impl_msg
-  def apply(cond: Bool): Unit = macro apply_impl  // macros currently can't take default arguments
+  // Macros currently can't take default arguments, so we need two functions to emulate defaults.
+  def apply(cond: Bool, message: String): Unit = macro AssertTransform.apply_msg
+  def apply(cond: Bool): Unit = macro AssertTransform.apply
 
-  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree): c.Tree = {
+  def do_apply(cond: Bool, message: String)(sourceInfo: SourceInfo): Unit = macro apply_impl_msg
+  def do_apply(cond: Bool)(sourceInfo: SourceInfo): Unit = macro apply_impl
+
+  def apply_impl_msg(c: Context)(cond: c.Tree, message: c.Tree)(sourceInfo: c.Tree): c.Tree = {
```

Comment:
Just Assert is so complicated now:
You call assert, this via AssertTransform macro this transform into do_apply.
Then, you try to call do_apply, this via the local macro transforms apply_impl.
And, depending on implicit resolution scope, it seems you could get a call to the sourceinfo materialize macro.

Finally, as a bonus, there are actually two separate macro expansion paths depending on whether there is a message or not.